src/cookbook/work-with-non-mintable-assets.md

```diff
@@ -0,0 +1,58 @@
+---
+title: "Work with Non-Mintable Assets | Cookbook"
+head:
+  - - meta
+    - name: description
+      content: "Learn how to work with non-mintable assets in Iroha."
+  - - meta
+    - name: keywords
+      content: "non-mintable assets, Iroha assets"
+---
+
+# How to Work with Non-Mintable Assets
+
+```rust
+fn register_non_mintable_asset(
+    iroha: &Client,
+) {
+    let magical_keys = AssetDefinitionId::from_str("magical_key#wonderland").unwrap();
+    // register keys as an asset definition
+    let register_keys_as_a_concept = Register::asset_definition(
+        AssetDefinition::new(
+            magical_keys.clone(),
+            AssetValueType::Numeric(NumericSpec::integer()),
+        ).mintable_once()
+    );
+    let alice = "alice@wonderland".parse().unwrap();
+    // Alice owns ten keys and cannot mint more
+    let initial_keys_of_alice = Asset::new(
+        AssetId::new(roses, alice), 
+        10_u32
+    );
+    let register_keys_of_alice = Register::asset(initial_keys_of_alice);
+    iroha.submit_all([
+        InstructionBox::from(register_keys_as_a_concept),
+        InstructionBox::from(register_keys_of_alice)
+    ]).unwrap();
```

Comment:
```suggestion
    iroha.submit_all([register_keys_as_a_concept, register_keys_of_alice]).unwrap();
```

this works because both instructions are of the same type

Reply:
The generics are different, one is `Register<AssetDefinition>`, the other is `Register<Asset>`. However, we can change `InstructionBox` to `RegisterBox`

Reply:
ok, then:
```suggestion
    let instructions: [InstructionBox; _] = [
        register_keys_as_a_concept.into(),
        register_keys_of_alice.into()
    ];
    iroha.submit_all(instructions).unwrap();
```